[ES|QL] Enables the addition of columns from the doc viewer

stratoula opened the pull request:
## Summary

From our user sessions we found out that many users on the ES|QL mode they are adding columns to the Document view from the Document viewer but the ES|QL mode is missing this functionality. This PR enables this action. We haven't decided what we are going to do with filtering yet so I disable all the filtering related actions.

<img width="1886" alt="image" src="https://github.com/elastic/kibana/assets/17003240/61f93e53-c1e4-4581-bc89-41c8422ef4f3">

### Checklist

- [ ] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios

elasticmachine commented:
Pinging @elastic/kibana-data-discovery (Team:DataDiscovery)

ninoslavmiskovic commented:
🏃‍♀️ 😀 Nice !

davismcphee reviewed (commented):
Changes look good and work well overall, just a few of points of feedback:

The "Actions" column now shows in the Single Document page when it used to be hidden:
<img width="932" alt="single_doc" src="https://github.com/elastic/kibana/assets/25592674/a3d81dd2-6e43-4ba7-b919-148c4cddb9fb">

The "Pin field" functionality is disabled for ES|QL even though it's not a filter function, it just allows users to pin certain fields to the top of the doc viewer as they page through docs. It might be worth supporting this for ES|QL too:
<img width="637" alt="pin_field" src="https://github.com/elastic/kibana/assets/25592674/cd878712-cb78-4400-9781-372c5c776d19">

The actions still seem to be hidden when using the legacy grid. Not sure if we care about supporting this TBH, so I'll leave it up to you to decide:
<img width="912" alt="inline" src="https://github.com/elastic/kibana/assets/25592674/e597b51a-98dc-4afb-92bd-2bcd8e7761c9">

stratoula commented:
Thanx @davismcphee for your thorough review. 

- I fixed the surrounding docs issue
- I enabled the field pin (I read it as filter pin tbh :D and this is why I disabled it)
- I added this to the legacy table too. Not so important but it was an easy addition.

davismcphee reviewed (approved):
The latest change LGTM, and I tested each scenario again and they're all working as expected now 👍 Thanks!
